src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -573,19 +598,25 @@ private static void waitForJobCompletion(BigQueryHelper bigQueryHelper, String p
               numOfErrors = errors.size();
             }
             // Only add first error message in the exception. For other errors user should look at BigQuery job logs.
-            throw new IOException(String.format("Error occurred while importing data to BigQuery '%s'." +
-                                                  " There are total %s error(s) for BigQuery job %s. Please look at " +
-                                                  "BigQuery job logs for more information.",
-                                                errorMessage, numOfErrors, jobReference.getJobId()));
+            String errorMessageException = String.format("Error occurred while importing data to BigQuery '%s'." +
+                " There are total %s error(s) for BigQuery job %s. Please look at " +
+                "BigQuery job logs for more information.",
+              errorMessage, numOfErrors, jobReference.getJobId());
+            throw ErrorUtils.getProgramFailureException(
+              new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN), errorMessageException,
+              String.format(errorMessageFormat, ErrorPhase.COMMITTING, errorMessageException), ErrorType.SYSTEM, true,
```

Comment:
I think error type here should be `UNKNOWN`, how are we sure it is a `system` issue?

Reply:
What metric can be used to better distinguish between system / unknown error type ?

Here the job failed on BigQuery "System" since the logs are pulled in from the BigQuery, so the reason will likely be known.

Reply:
Job can fail on BigQuery due to many reasons like permission issues as well right? So, we need to decide based on error codes & error reason cannot directly say that it is a system error.

Reply:
Updated to type `UNKNOWN`